src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,25 @@
+fun main(args: Array<String>) {
+    scenario(12,8)
+}
+
+fun scenario(numberOfCars : Int, capacity : Int) {
+    /*
+    * Creates a given a number of cars, every other car wants to drive
+    * A road network is created with the given capacity
+    * The road network is analyzed and it is printed whether a car is delayed or not
+    */
+    val listOfCars = ListOfCars()
+    val road = Network(capacity,listOfCars)
+    for (i in 1..numberOfCars) {
+        if (i % 2 == 0) {
+            Car(i,true, listOfCars)
+        } else {
+            Car(i,false, listOfCars)
+        }
+    }
+    road.analyzeNetwork()
+    for(car in road.listOfCarsDriving) {
```

Comment:
Because only the cars who want to drive can be delayed, or should I iterate over the initial list and ask there if they want to drive (if)?